### DISCUSSION TITLE
Best place/way to publish a new pedal based on the Hothouse?

### DISCUSSION THREAD

**Discussion by joulupukki** [clevelandmusicco/HothouseExamples#14] [Q&A] [answered]
I've been working on a version of the pedal that combines reverb, tremolo, and delay. Is there a recommended way of publishing it?

One of the things I did is to port the Spoodge Reverb as a new daisysp class so it can more easily be used, which my new pedal code uses.

Do I submit it to the Examples repo? Do I create my own repo to house it? If I publish it on my own, is there a recommended way of including the Hothouse libraries/etc?

**Comment by clevelandmusicco** [comment]
Great question! The short answer is: it's up to you. I welcome pull requests for this repo as long as they follow the conventions already established:

1. Fork this repo.
2. Use the `create_new_proj.py` helper script to create your effect's scaffolding in your fork, and then edit as needed.
3. Edit the `README.md` with relevant info about your effect.
4. Submit a pull request back to this repo.

Here's an example of a user-submitted effect: https://github.com/clevelandmusicco/HothouseExamples/tree/main/src/TremVerb.

If you port third party libraries, keep them in your effect's source directory. For example: `./src/MyAwesomeEffect/SomePortedLibrary/<third party files here>`. Just make sure your Makefile works.

I honestly think this is the easiest way to work with the Hothouse and create your own effects. But ...

If you decide to go off-road and publish your own project, the structure of your porject and how you include the Hothouse stuff is totally up to you. There are jut a few needed `hothouse.*` files in the `src` directory. Just be nice and link back here, and let me know so I can add your project to our user-submitted effects list 😉

**Comment by joulupukki** [comment]
Thanks for answering. Yeah, i guess it’s just a matter of how to include other third-party items. In this case, it’s a “Sploodge” Reverb: https://github.com/joulupukki/hothouse-effects/blob/main/Flick/reverbsploodge.cpp

I’m not sure what to do with choosing the right license/etc. Sploodge Reverb isn’t probably what I’ll land on for the Flick project, but it’s at least a start.

One thing I have in the Flick project that could be interesting to incorporate into the Hothouse hardware files is supporting double press. If you look at flick.cpp you’ll notice that I can turn on and off both the delay and the tremolo independently using a single foot switch. The Tremolo with a double press and the Delay with a single press. I didn’t spend enough time to figure out how to get that in the Hothouse main library, but it’d be a nice built-in feature.

**Reply by clevelandmusicco** [reply, marked as answer]
The Flick looks pretty cool and would definitely be a fit in the Hothouse repo! I also looked at your `hothouse-effects` and it would be really easy to move the `/Flick` subdirectory into a fork of the Hothouse repo so you could submit a pull request: just follow your own installation instructions in the fork and you're off to the races. If you're not familiar with PRs, check out https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork. 

As for the double-tap and single-tap functionality, I totally agree it could be a nice addition to the Hothouse "library". We'd have to think through how best to structure the development-time UX so developers could assign code to actions (callbacks, lambdas, some IoC approach, etc.) But in any case, adding that sort of thing in a fork and submitting a PR would be the right approach there as well.

The whine you mention in your video seems to be hit-or-miss for folks. Many Hothouse users say they have no issues until they add other pedals to the signal chain (which might point to power supply issues). Others say they can't reproduce it at all. On my test rig, there is no whine with any of the examples in the repo.

One suggestion would be to experiment with sample rate / blocksize in the code. Scroll through this: https://forum.electro-smith.com/t/questions-about-digital-noise-and-grounding/432/10. I suspect you could get away with a much smaller blocksize in your Flick code.

**Reply by joulupukki** [reply]
Thanks for all the good info. I’ve updated my project to use submodules for libDaisy, DaisySP, and another plate reverb. I pulled in hothouse.cpp in there for now and once I get it fully working, maybe that’s the point when I do a pull request into the examples. We’ll see how it goes.

I’m currently trying to get this in there as the reverb: https://github.com/digitalartifactmusic/PlateauNEVersio

**Reply by joulupukki** [reply]
Oh, the previous project link doesn’t quite work. Here’s an updated link to the root of the project and the instructions to get it working (similar to the HH Examples): https://github.com/joulupukki/hothouse-effects/tree/main?tab=readme-ov-file#installation
